lib/Sema/ConstraintSystem.h

```diff
@@ -3478,25 +3527,22 @@ class TypeVarBindingProducer {
 public:
   TypeVarBindingProducer(ConstraintSystem &cs, TypeVariableType *typeVar,
                          ArrayRef<Binding> initialBindings)
-      : CS(cs), TypeVar(typeVar),
+      : BindingProducer(cs, typeVar->getImpl().getLocator()), TypeVar(typeVar),
         Bindings(initialBindings.begin(), initialBindings.end()) {}
 
-  Optional<TypeVariableBinding> operator()() {
+  std::unique_ptr<TypeBinding> operator()() override {
     // Once we reach the end of the current bindings
     // let's try to compute new ones, e.g. supertypes,
     // literal defaults, if that fails, we are done.
     if (needsToComputeNext() && !computeNext())
-      return None;
+      return nullptr;
 
-    return TypeVariableBinding(TypeVar, Bindings[Index++]);
+    unsigned currIndex = Index++;
+    return llvm::make_unique<TypeVariableBinding>(currIndex, TypeVar,
```

Comment:
@rudkx For some reason solver would just crash when I tried to do `new (CS.getAllocator())`, I also tried to call
```c++
void *mem = CS.getAllocator().Allocate(sizeof(TypeVariableBinding), alignof(TypeVariableBinding)); 
auto *binding = new (mem) ...
```

which didn't help either... It seems to work fine with `new` though... Any ideas?

Reply:
Does it crash when allocating, or later?

Reply:
Later, when it’s trying to access fields I think